### DISCUSSION TITLE
Way to have Neovide honor `NVIM_APPNAME` environment variable?

### DISCUSSION THREAD

**Discussion by proofer** [neovide/neovide#2631] [Q&A] [answered]
nvim uses the value of NVIM_APPNAME as the directories it will use for config and state files instead of the default `.../nvim/`; this makes it easy to experimentally switch configs. But it doesn't seem to be honored in Neovide. As yet I don't know of a way to have Neovide's nvim use the value. Because Neovide is my "daily driver" it would be good to check a prospective config in it.

**Comment by fredizzimo** [comment]
Which os are you using?

`NVIM_APPNAME=foo neovide` works just fine for me on Linux.

**Comment by proofer** [comment]
MacOS
In zsh:
`export NVIM_APPNAME=foo`
then I click Neovide's dock icon.

**Reply by fredizzimo** [reply]
The environent variables from the shell won't be exported to the dock like that. You need to start neovide from the same shell after exporting it, or put it somewhere like the `.zshenv` or `.zprofile`, which are sourced during startup. 

Alternatively, you can have some startup script for the dock, but I'm not a mac user, so I can't help you with that. @falcucci probably can. I know that @9mm, also has set up something similar.

**Reply by 9mm** [reply, marked as answer]
@proofer so to handle ENV variables with the dock icon (which i use much less frequently, but still use it via drag and drop for CSVs and stuff), i put this in my `~/.zshrc`

```bash
# Neovide
# Instead of using launchd properly, just put them here so we don't forget
launchctl setenv NEOVIDE_FRAME "buttonless"
launchctl setenv NEOVIDE_TITLE_HIDDEN "true"
launchctl setenv NEOVIDE_VSYNC "0"
launchctl setenv NEOVIDE_FORK "0"
```

Technically theres a better way to do this where you run these commands once, but i decided to run them here instead that way i can commit them to my dotfiles, etc... otherwise i will forget where they all are and so on.

**Reply by proofer** [reply]
Interesting, @9mm   ...Not only interesting, but it works!
Thanks to you and @fredizzimo

**Reply by 9mm** [reply]
sweet! this is if you want to persist across reboots: https://www.dowdandassociates.com/blog/content/howto-set-an-environment-variable-in-mac-os-x-slash-etc-slash-launchd-dot-conf/
